react/src/components/Files/FileListing.tsx

```diff
@@ -87,33 +97,98 @@ export const FileListing: React.FC<FileListingProps> = ({
 
   const selectedSystemId = watch('system');
 
+  const setRootFolderChain = (rootPath, sys = selectedSystem) => {
+    let rootFolderName: string;
+
+    if (sys.id === myDataSystem?.id) {
+      rootFolderName = 'My Data';
+    } else if (sys.id === communityDataSystem?.id) {
+      rootFolderName = 'Community Data';
+    } else if (sys.id === publishedDataSystem?.id) {
+      rootFolderName = 'Published Data';
```

Comment:
```suggestion
    if (sys?.id === myDataSystem?.id) {
      rootFolderName = 'My Data';
    } else if (sys?.id === communityDataSystem?.id) {
      rootFolderName = 'Community Data';
    } else if (sys?.id === publishedDataSystem?.id) {
      rootFolderName = 'Published Data';
```
for fixing unit tests.  
or check if sys is null as the first `if`? 🤔